incident_viewer_demo/incident/incident.go

```diff
@@ -65,74 +64,45 @@ func (i *IncidentData) Init(
+	i.BadPeriodEnd = badTimeEnd
 	i.GoodPeriodMetric = goodMetric
 	i.BadPeriodMetric = badMetric
-	i.BadPeriodEnd = badTimeEnd
 	i.Severity = severity
 	i.NumTestsAffected = testsAffected
 	i.GoodPeriodInfo = goodPeriodInfo
 	i.BadPeriodInfo = badPeriodInfo
 	i.IncidentInfo = incidentInfo
 }
 
-func (i *DefaultIncident) Init(goodTimeStart time.Time, goodTimeEnd time.Time,
+/* Assign data members and set appropriate text for information fields */
+func (i *DefaultIncident) MakeIncidentData(goodTimeStart time.Time, goodTimeEnd time.Time,
 	badTimeStart time.Time,
 	badTimeEnd time.Time,
 	avgDSGood float64,
 	avgDSBad float64,
-	severity float64,
+	severityDecimal float64,
 	testsAffected int) {
 
+	gds := strconv.FormatFloat(avgDSGood, 'f', 2, 64)
+	bds := strconv.FormatFloat(avgDSGood, 'f', 2, 64)
+	s := strconv.FormatFloat(severityDecimal*100, 'f', 2, 64)
+	ta := strconv.Itoa(testsAffected)
+
 	i.goodStartTime = goodTimeStart
 	i.goodEndTime = goodTimeEnd
 	i.badStartTime = badTimeStart
 	i.badEndTime = badTimeEnd
 	i.avgGoodDS = avgDSGood
 	i.avgBadDS = avgDSBad
-	i.severityDecimal = severity
+	i.severity = severityDecimal
 	i.numTestsAffected = testsAffected
-}
-
-//**************************************//
-//    		GETTER METHODS
-//**************************************//
-
-func (i *DefaultIncident) GetGoodPeriod() (time.Time, time.Time) {
-	return i.goodStartTime, i.goodEndTime
-}
-
-func (i *DefaultIncident) GetBadPeriod() (time.Time, time.Time) {
-	return i.badStartTime, i.badEndTime
-}
-
-func (i *DefaultIncident) GetGoodMetric() float64 {
-	return i.avgGoodDS
-}
-
-func (i *DefaultIncident) GetBadMetric() float64 {
-	return i.avgBadDS
-}
-
-func (i *DefaultIncident) GetSeverity() float64 {
-	return i.severityDecimal
-}
-
-func (i *DefaultIncident) GetTestsAffected() int {
-	return i.numTestsAffected
-}
-
-func (i *DefaultIncident) GetGoodPeriodInfo() string {
-	ds := strconv.FormatFloat(i.avgGoodDS, 'f', 2, 64)
-	return "Average download speed: " + ds + " mb/s"
-}
+	i.goodPeriodInfo = "Average download speed: " + gds + " mb/s"
+	i.badPeriodInfo = "Average download speed: " + bds + " mb/s"
+	i.incidentInfo = "Download speed dropped by " + s + "% affecting " + ta + " tests"
 
-func (i *DefaultIncident) GetBadPeriodInfo() string {
-	ds := strconv.FormatFloat(i.avgBadDS, 'f', 2, 64)
-	return "Average download speed: " + ds + " mb/s"
 }
 
-func (i *DefaultIncident) GetIncidentInfo() string {
-	s := strconv.FormatFloat(i.severityDecimal*100, 'f', 2, 64)
-	ta := strconv.Itoa(i.numTestsAffected)
-	return "Download speed dropped by " + s + "% affecting " + ta + " tests"
+/* Retrieve all Incident fields */
+func (i *DefaultIncident) GetIncidentData() (time.Time, time.Time, time.Time, time.Time, float64, float64, float64, int, string, string, string) {
+	return i.goodStartTime, i.goodEndTime, i.badStartTime, i.badEndTime, i.avgGoodDS, i.avgBadDS, i.severity, i.numTestsAffected, i.goodPeriodInfo, i.badPeriodInfo, i.incidentInfo
```

Comment:
are the data members public? :o

Reply:
or is it because you're still in the same file?

Reply:
The data members of the DefaultIncident are private and yes they can be accessed here because it is in the same package I believe.